Error enabling LogicGates v1.1.8 World Edit not found

[17:12:15 INFO]: [bootstrap] Running Java 21 (OpenJDK 64-Bit Server VM 21.0.5+11; Red Hat, Inc. (Red_Hat-21.0.5.0.11-1)) on Linux 6.12.9-100.fc40.x86_64 (amd64)

[17:12:21 INFO]: This server is running Paper version 1.21.4-114-main@cd9d6d6 (2025-01-15T11:21:28Z) (Implementing API version 1.21.4-R0.1-SNAPSHOT)


17:12:27 ERROR]: Error occurred while enabling LogicGates v1.1.8 (Is it up to date?)
java.lang.NoClassDefFoundError: com/sk89q/worldedit/WorldEdit
	at LogicGates-1.1.8.jar/pl.bednarskiwsieci.logicgatesplugin.LogicGatesPlugin.onEnable(LogicGatesPlugin.java:126) ~[LogicGates-1.1.8.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:284) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:202) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:109) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:520) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.CraftServer.enablePlugin(CraftServer.java:656) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at org.bukkit.craftbukkit.CraftServer.enablePlugins(CraftServer.java:605) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:743) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:488) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:322) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1163) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:310) ~[paper-1.21.4.jar:1.21.4-114-cd9d6d6]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.sk89q.worldedit.WorldEdit
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:197) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:164) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
	... 13 more
[17:12:27 INFO]: [LogicGates] Disabling LogicGates v1.1.8
[17:12:27 INFO]: [LogicGates] Version 1.1.8 (API 1.16) Disabled - made by Piotr Bednarski

--- a/build.gradle
+++ b/build.gradle
@@ -4,7 +4,7 @@ plugins {
 }
 
 group = 'pl.bednarskiwsieci'
-version = '1.1.7'
+version = '1.1.9'
 
 repositories {
     mavenCentral()
@@ -24,8 +24,9 @@ repositories {
 
 dependencies {
     compileOnly("io.papermc.paper:paper-api:1.20.1-R0.1-SNAPSHOT")
-    compileOnly("com.sk89q.worldedit:worldedit-bukkit:7.3.0")
     compileOnly("com.google.code.gson:gson:2.12.1")
+    compileOnly("com.sk89q.worldedit:worldedit-bukkit:7.3.0")
+    compileOnly("com.sk89q.worldedit:worldedit-core:7.3.0")
     implementation("org.bstats:bstats-bukkit:3.0.0")
 }
 

--- a/src/main/java/pl/bednarskiwsieci/logicgatesplugin/LogicGatesPlugin.java
+++ b/src/main/java/pl/bednarskiwsieci/logicgatesplugin/LogicGatesPlugin.java
@@ -1,15 +1,7 @@
 package pl.bednarskiwsieci.logicgatesplugin;
 
-import pl.bednarskiwsieci.logicgatesplugin.commands.LogicGatesCommand;
-import pl.bednarskiwsieci.logicgatesplugin.integrations.WorldEditIntegration;
-import pl.bednarskiwsieci.logicgatesplugin.listeners.GateListener;
-import pl.bednarskiwsieci.logicgatesplugin.model.GateData;
-import pl.bednarskiwsieci.logicgatesplugin.model.GateType;
-import pl.bednarskiwsieci.logicgatesplugin.util.ConfigManager;
-import pl.bednarskiwsieci.logicgatesplugin.util.GateUtils;
-import pl.bednarskiwsieci.logicgatesplugin.util.GatesConfigManager;
-import pl.bednarskiwsieci.logicgatesplugin.util.UpdateChecker;
 import com.sk89q.worldedit.WorldEdit;
+import com.sk89q.worldedit.util.eventbus.EventBus;
 import org.bstats.bukkit.Metrics;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
@@ -25,8 +17,18 @@ import org.bukkit.block.data.type.RedstoneWire;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
 import org.bukkit.scheduler.BukkitTask;
+import pl.bednarskiwsieci.logicgatesplugin.commands.LogicGatesCommand;
+import pl.bednarskiwsieci.logicgatesplugin.integrations.WorldEditIntegration;
+import pl.bednarskiwsieci.logicgatesplugin.listeners.GateListener;
+import pl.bednarskiwsieci.logicgatesplugin.model.GateData;
+import pl.bednarskiwsieci.logicgatesplugin.model.GateType;
+import pl.bednarskiwsieci.logicgatesplugin.util.ConfigManager;
+import pl.bednarskiwsieci.logicgatesplugin.util.GateUtils;
+import pl.bednarskiwsieci.logicgatesplugin.util.GatesConfigManager;
+import pl.bednarskiwsieci.logicgatesplugin.util.UpdateChecker;
 
 import java.io.File;
 import java.util.*;
@@ -124,12 +126,19 @@ public class LogicGatesPlugin extends JavaPlugin {
             getLogger().severe("An error occurred while checking for updates");
         }
 
-        // Register WorldEdit Integration
-        try {
-            WorldEdit.getInstance().getEventBus().register(new WorldEditIntegration(this));
-            getLogger().info("WorldEdit Integration has been registered");
-        } catch (Exception e) {
-            getLogger().severe("Failed to register WorldEdit integration");
+        // Register WorldEdit Integration if WorldEdit is present
+        if (isWorldEditPresent()) {
+            try {
+                WorldEdit worldEditInstance = WorldEdit.getInstance();
+                EventBus eventBus = worldEditInstance.getEventBus();
+                eventBus.register(new WorldEditIntegration(this));
+                getLogger().info("WorldEdit Integration has been registered");
+            } catch (Exception e) {
+                getLogger().severe("Failed to register WorldEdit integration");
+                e.printStackTrace();
+            }
+        } else {
+            getLogger().info("WorldEdit not found, skipping integration");
         }
     }
 
@@ -158,6 +167,16 @@ public class LogicGatesPlugin extends JavaPlugin {
     //endregion
 
     //region Component Registration
+    private boolean isWorldEditPresent() {
+        Plugin[] plugins = Bukkit.getPluginManager().getPlugins();
+        for(Plugin plugin : plugins) {
+            if (plugin.getName().toLowerCase().contains("worldedit")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     private void registerCommands() {
         Objects.requireNonNull(getCommand("logicgates")).setExecutor(new LogicGatesCommand(this, configManager, updateChecker));
     }

